src/osd/OSD.cc

```diff
@@ -4623,6 +4629,36 @@ void OSD::handle_osd_ping(MOSDPing *m)
                      << " , erase pending ping(sent at " << m->stamp << ")"
                      << " and older pending ping(s)"
                      << dendl;
+	    // Record per osd interace ping times
+	    // Based on osd_heartbeat_interval ignoring that it is randomly short than this interval
+#define ROUND_S_TO_MSEC(sec) (int)((sec) * 1000 * 1000 + 0.5)
+	    if (hb_back_pingtime.size() < hb_vector_size) {
```

Comment:
@jdurgin @neha-ojha Thinking about this some more.  It would be better to keep network time per OSD instead of per network interface.  I didn't do this initially because of the storage requirements (256 samples * # OSDs * 2 interfaces * 4 bytes minimum) = 600K.  However, I could keep a 1 minute average and a circular buffer of the last 16.

Unfortunately, a randomization of the osd_heartbeat_interval (6 seconds by default) doesn't allow me just get 60 / osd_heartbeat_interval pings to be exactly 1 minute.  Also, we'll need to restrict the value of osd_heartbeat_interval to be no greater than 60.  [currently .set_min_max(1, 86400) a maximum of 1 per 24 hours]